parquet-hadoop/src/main/java/org/apache/parquet/hadoop/codec/SnappyDecompressor.java

```diff
@@ -34,6 +35,8 @@
   private ByteBuffer inputBuffer = ByteBuffer.allocateDirect(0);
 
   private boolean finished;
+
+  private int maxBufferSize = 64 * 1024 * 1024;
```

Comment:
Why was this introduced?

Reply:
Used for reset below.We can remove this.But i think for the purpose of this patch,we should also refactor reset api.
Any suggestion?Thanks @rdblue

Reply:
Can you elaborate more than "used for reset below"? What was the intent? Also, why do you think the reset API needs to be refactored?